python/ray/train/backend.py

```diff
@@ -19,6 +20,10 @@ class BackendConfig:
     def backend_cls(self):
         return Backend
 
+    @property
+    def train_func_context(self):
```

Comment:
Note: I think this should technically be the interface of a `DataParallelBackendConfig`, since it refers to the `train_loop_per_worker` context, which doesn't exist on the `BaseTrainer`.

But, soon every trainer implementation will be built on `DataParallelTrainer`, so this is not that important.

Reply:
`Backend` and `BackendConfig` are specific to `DataParallelTrainer` so this should be fine!

Reply:
:+1: